### ISSUE TITLE
Typescript Ably.Types.TokenParams interface type incorrect?

### ISSUE THREAD

**Issue by ekron** [ably/ably-js#585]
If I create an object of type `Ably.Types.TokenParams`, I expect to be able to do this:

```typescript
  const tokenParams: Ably.Types.TokenParams = {
    capability: { "*": ["subscribe"] },
    clientId: userId,
    // etc
  };
```

But I get an error:

```
Type '{ "*": string[]; }' is not assignable to type 'string'.ts(2322)
ably.d.ts(161, 3): The expected type comes from property 'capability' which is declared here on type 'TokenParams'
```

Is the `capability` field of the `Ably.Types.TokenParams` interface incorrect?

```typescript
interface TokenParams {
	capability?: string; // this seems to be incorrect?
	clientId?: string;
	nonce?: string;
	timestamp?: number;
	ttl?: number;
}
```

Or perhaps I have misunderstood something?

Thanks for your help.

**Comment by SimonWoolf** [comment]
You're right.

The reason it's declared as a `string` is because the typings were derived from the REST api type declaration ([TK2b](https://docs.ably.io/client-lib-development-guide/features/#TK2b)). But the client library abstracts that away by (optionally) stringifying the capability for you. So the correct typing would be something like `{ [key: string]: capabilityOp[]; } | string`, where `capabilityOp` is a union type of the permitted capability operations. I'll fix this now.

**Comment by ekron** [comment]
Thanks for the quick response, Simon, appreciate your help.

**Comment by SimonWoolf** [comment]
No problem. Fixed in 1.1.7, https://github.com/ably/ably-js/commit/dab9bb26d3d16ccd6c181f8d12100e069a41b26c
